### PR TITLE
Better function transport

### PR DESCRIPTION
**Pull request by jkanem**
- [ ] I have added the tests to cover my changes.
- [x] I have updated the documentation, VERSION, and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

- Fixes https://github.com/AgnostiqHQ/covalent-slurm-plugin/issues/11 (remote machine shouldn't need Covalent installed)
- `args` and `kwargs` are now pickled along with the function for transport to the remote machine. Previously, these two inputs were sent as strings, which sometimes failed.

### CODE REVIEW

File: covalent_slurm_plugin/slurm.py [hunk 1]
```diff
@@ -314,13 +309,13 @@ def _format_submit_script(
...
 with open("{func_filename}", "rb") as f:
-    function = pickle.load(f).get_deserialized()
+    function, args, kwargs = pickle.load(f)
 
 result = None
 exception = None
 
 try:
-    result = function(*{args}, **{kwargs})
+    result = function(*args, **kwargs)
```
Comment (venkatBala) [inline covalent_slurm_plugin/slurm.py:318]: For long `*args` and `**kwargs` this will need to be tested further

**Review by venkatBala — APPROVED** [review]
looks good!

**Comment by cjao** [comment]
The remote machine will indeed need the Covalent SDK installed as it needs to understand basic Covalent data types -- `Electron`, `Lattice`, `TransportableObject` -- but not the entire dispatcher.

**Comment by arosen93** [comment]
Ah, makes sense! Then I think the covalent docs + README will just need to be updated to prevent friction for new users (#48).

**Pull request merged**